aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -44,7 +45,38 @@ inline void _scale_attn_mask_fusion_kernel(
   }
   for (; i < size; i++) {
     auto tmp0 = a[i];
-    auto tmp1 = (T1) b[i];
+    auto tmp1 = (T1)b[i];
+    out[i] = tmp0 * val + tmp1;
+  }
+}
+
+// out = val * a + b for b_stride=0
+template <typename T1, typename T2>
+inline void _scale_attn_mask_fusion_kernel_stride0(
```

Comment:
Consider to reuse the `_scale_attn_mask_fusion_kernel`, .e.g. templaterizing it, to support stride0 case? Most of the logic is same.

Reply:
Modified to use the template parameter for stride0 case.